File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res=0;
    
    void count(TreeNode* root, int maxi){
        if(!root)return;
        if(root->val>=maxi){
            res++;
            maxi=root->val;
        }
        count(root->left,maxi);
        count(root->right,maxi);
    }
    
    int goodNodes(TreeNode* root) {
        count(root,-1e5);
        return res;
    }
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def count(root, maxi):
            if root==None:
                return 0
            
            good=0
            if root.val>=maxi:
                good=1
            maxi=max(root.val,maxi)
            
            return good+count(root.left,maxi)+count(root.right,maxi)
            
        return count(root,-100000)
        